src/modules/job-manager/event.c

```diff
@@ -200,6 +252,8 @@ void event_batch_destroy (struct event_batch *batch)
                                batch->annotations);
             json_decref (batch->annotations);
         }
+        if (batch->events_to_send)
+            send_events_to_listeners (batch->event);
         if (batch->responses) {
             flux_msg_t *msg;
             flux_t *h = batch->event->ctx->h;
```

Comment:
I was initially slightly confused by this code, since `send_events_to_listeners()` is sandwiched between code that publishes events, and code that sends deferred responses.  It might be good to:

* rename `events_to_send` flag to something like `listener_response_available`
* rename  `send_events_to_listeners()` to something like `generate_listener_responses()`
* have that function call `event_batch_respond()` to create response messages and append to `batch->responses`, and then let the next block of code send the responses instead of repeating it

Also the comment block should be updated at the top of `event_batch_destroy()`.

Reply:
ahhh, i did not know about the event batch responses